File: CS/WindowsApplication3/Main.cs
using DevExpress.XtraEditors;
using System;
using System.Data;
using System.Drawing;


namespace DXSample
{
    public partial class Main : XtraForm
    {
        public Main()
        {
            InitializeComponent();
        }

        DataTable GetCategoriesDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Categories";
            table.Columns.Add(new DataColumn("CategoryID", typeof(int)));
            table.Columns.Add(new DataColumn("CategoryName", typeof(string)));
            table.Columns.Add(new DataColumn("Picture", typeof(Image)));
            Random random = new Random();
            for(int i = 0; i < 20; i++)
            {
                int index = i + 1;
                Image image = DXSample.Properties.Resources.about_32x32;
                if(random.Next(0, 2) == 0) image = DXSample.Properties.Resources.convert_32x32;
                table.Rows.Add(index, "Category " + index, image);
            }
            return table;
        }

        private void OnFormLoad(object sender, EventArgs e)
        {
            categoriesBindingSource.DataSource = GetCategoriesDataTable();
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports DevExpress.XtraEditors
Imports System
Imports System.Data
Imports System.Drawing


Namespace DXSample
    Partial Public Class Main
        Inherits XtraForm
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Function GetCategoriesDataTable() As DataTable
            Dim table As DataTable = New DataTable()
            table.TableName = "Categories"
            table.Columns.Add(New DataColumn("CategoryID", GetType(Integer)))
            table.Columns.Add(New DataColumn("CategoryName", GetType(String)))
            table.Columns.Add(New DataColumn("Picture", GetType(Image)))
            Dim random As Random = New Random()

            For i As Integer = 0 To 20 - 1
                Dim index As Integer = i + 1
                Dim image As Image = Resources.about_32x32
                If random.[Next](0, 2) = 0 Then image = Resources.convert_32x32
                table.Rows.Add(index, "Category " & index, image)
            Next

            Return table
        End Function

        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            categoriesBindingSource.DataSource = GetCategoriesDataTable()
        End Sub
    End Class
End Namespace
